scripts/operations/validation/ncn111.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+
+main() {
+    if check_velero; then
+        velero_count=1
+    else
+        velero_count=0
+    fi
+
+    if check_spire_agent; then
+        spire_count=1
+    else
+        spire_count=0
+    fi
+
+    if cfs_state_reporter; then
+        cfs_count=1
+    else
+        cfs_count=0
+    fi
+
+    if clock_skew; then
+        clock_count=1
+    else
+        clock_count=0
+    fi
+
+    count_success=$(($velero_count + $spire_count + $cfs_count + $clock_count))
+    count_total=4
+    count_failed=$(($count_total - $count_success))
+
+    echo " "
+    echo "======================================="
+    echo " "
+    echo "Summary: $count_failed of $count_total tests FAILED"
+    if [[ "$velero_count" == 0 ]]; then
+        echo " "
+        echo "- Velero failed backups found, proceed to https://github.com/Cray-HPE/docs-csm/blob/CASMINST-5064/troubleshooting/known_issues/issues_with_ncn_health_checks.md for remediation steps"
+        echo " "
+    fi
+
+    if [[ "$spire_count" == 0 ]]; then
+        echo "- spire-agent is not enabled please follow documentation at https://github.com/Cray-HPE/docs-csm/blob/CASMINST-5064/troubleshooting/known_issues/issues_with_ncn_health_checks.md"
+        echo " "
+    fi
+
+    if [[ "$cfs_count" == 0 ]]; then
+        echo "- Issues found on CFS, please proceed to https://github.com/Cray-HPE/docs-csm/blob/CASMINST-5064/troubleshooting/known_issues/issues_with_ncn_health_checks.md"
+        echo " "
+    fi
+
+    if [[ "$clock_count" == 0 ]]; then
+        echo "- Issues found on clock, please proceed to https://github.com/Cray-HPE/docs-csm/blob/CASMINST-5064/troubleshooting/known_issues/issues_with_ncn_health_checks.md"
+        echo " "
+    fi
+}
+
+check_velero() {
+    echo "Checking for any Velero failed backups"
+    sleep 2
+    velero=$(kubectl get backups -A -o json | jq -e '.items[] | select(.status.phase == "PartiallyFailed") | .metadata.name')
+    if [[ $? != 0 ]]; then
+        echo "No issues found."
+        echo "---------------------------------------"
+        return 0
+    else
+        echo " "
+        echo "##ERROR Found failed backup ##"
+        kubectl get backups -A -o json | jq -e '.items[] | select(.status.phase == "PartiallyFailed") | .metadata.name' | sed -e 's/^"//' -e 's/"$//'
+        echo "---------------------------------------"
+        return 1
+    fi
+}
+
+check_spire_agent() {
+    echo "Checking spire agent status"
+    spire=$(kubectl get pods -n spire | grep request-ncn-join-token)
+    if [[ $? = 0 ]]; then
+        echo "No issues found."
+        echo "---------------------------------------"
+        return 0
+    else
+        echo "Issues found"
```

Comment:
I think it would be really helpful if you pipe the commands output to a file, then either cat the file here or reference it in the message above, instead of simply saying "Issues found".

This would mean that, for remediation, the user does not have to re-run this command to figure out what actually is wrong. The remediation doc becomes much smaller then, and we can even skip remediation automation as it is lower priority then validation (because validation runs 100% of the time, remediation runs <100% of the time).

This comment applies to all future validation scripts.